src/com/nkita/practice/collections/PersonStatisticHandler.java

```diff
@@ -298,5 +298,71 @@ public void printInfo(List<Person> personList) {
 
     }
 
+    public void checkByGenderMap() {
+        List<String> male = new ArrayList<>();
+        List<String> female = new ArrayList<>();
+
+        for (Person person : persons) {
+            if (person.getGender() == Gender.MALE) {
+                male.add(person.getName());
+            } else {
+                female.add(person.getName());
+            }
+        }
+
+        Map<Gender, List<String>> genderListMapMale = new HashMap<>();
+        Map<Gender, List<String>> genderListMapFemale = new HashMap<>();
```

Comment:
You do not need two maps here, because you can store all data in one with different keys. That's what they're designed for..